server/pulp/server/async/scheduler.py

```diff
@@ -317,15 +320,60 @@ def tick(self):
         :return:    number of seconds before the next tick should run
         :rtype:     float
         """
-        ret = super(Scheduler, self).tick()
         self._failure_watcher.trim()
-
         # this is not an event that gets sent anywhere. We process it
         # immediately.
         scheduler_event = {'timestamp': time.time(),
                            'type': 'scheduler-event',
-                           'hostname': ("%s@%s" % (SCHEDULER_WORKER_NAME, platform.node()))}
+                           'hostname': ("%s@%s@%s" % (SCHEDULER_WORKER_NAME,
+                                                      str(celery_name), platform.node()))}
         worker_watcher.handle_worker_heartbeat(scheduler_event)
+
+        celerybeat_name = SCHEDULER_WORKER_NAME + "@" + str(celery_name) + "@" + platform.node()
+
+        # Establishing connection with DB
+        db = db_connection.get_database()
+        collection = db['celerybeat_lock']
+
+        # Creating a unique constraint on the collection
+        collection.create_index('lock', unique=True)
```

Comment:
Don't do this here. Instead you should add your CeleryBeatLock object to the list of things that pulp-manage-db will call. Do that here:  https://github.com/pulp/pulp/blob/7a1703ceb1913c757c1f91caabd3ed44d458a7c5/server/pulp/server/db/manage.py#L123